core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedElementNameUtil.java

```diff
@@ -46,6 +47,16 @@ static String findName(Param param, Object nameRetrievalTarget) {
         return getName(nameRetrievalTarget);
     }
 
+    static String findName(Attribute attribute, Object nameRetrievalTarget) {
```

Comment:
Can we deduplicate two `findName()` methods by using method references? e.g.

```java
findName(nameRetrievalTarget, attribute::value);
findName(nameRetrievalTarget, param::value);
```

Reply:
I see! i modified it! 
IMHO, `static String findName(Header header, Object nameRetrievalTarget)` can be removed if we change method signature such like findName(Object nameRetrievalTarget, String value, Function<T, R> delegate). 

```java
findName(nameRetrievalTarget, param.value(), (r) -> r);
findName(nameRetrievalTarget, attribute.value, (r) -> r);
findName(nameRetrievalTarget, attribute.value, (r) -> toHeader(r));

....
    static String findName(Object nameRetrievalTarget, String value, Function <T,R> delegate) {
        requireNonNull(nameRetrievalTarget, "nameRetrievalTarget");

        final String value = attribute.value();
        if (DefaultValues.isSpecified(value)) {
            checkArgument(!value.isEmpty(), "value is empty.");
            return value;
        }
        
        return delegate.apply(getName(nameRetrievalTarget));
    }
```

But, I'm not sure how effective it is to remove method `static String findName(Header header, Object nameRetrievalTarget)` as well.

Reply:
Yeah, that was actually close to what I suggested :smile: We might want to provide one shortcut method though:

```java
findName(Object nameRetrievalTarget, String value) {
    return findName(nameRetrievalTarget, value, Function.identity());
}
```

Reply:
Aye, i updated it.